Rename libsdl packages to libsdl2

SirLynix opened the pull request:
Now that #6225 is merged, the next step is to rename libsdl package to libsdl2

SirLynix commented:
```
Run xmake l ./scripts/test.lua -D -a arm64-v8a -k shared -m debug
{ 
  "blah",
  "centurion",
  "cimgui",
  "fluidsynth",
  "imgui",
  "johnnyengine",
  "libsdl",
  "libsdl2",
  "libsdl2 2.0.8",
  "libsdl2 2.0.12",
  "libsdl2 2.0.14",
  "libsdl2 2.0.16",
  "libsdl2 2.0.18",
  "libsdl2 2.0.20",
  "libsdl2 2.0.22",
  "libsdl2 2.24.0",
  "libsdl2 2.24.2",
  "libsdl2 2.26.0",
  "libsdl2 2.26.1",
  "libsdl2 2.26.2",
  "libsdl2 2.26.3",
  "libsdl2 2.26.4",
  "libsdl2 2.26.5",
  "libsdl2 2.28.0",
  "libsdl2 2.28.1",
  "libsdl2 2.28.2",
  "libsdl2 2.28.3",
  "libsdl2 2.28.4",
  "libsdl2 2.28.5",
  "libsdl2 2.30.0",
  "libsdl2 2.30.1",
  "libsdl2 2.30.2",
  "libsdl2 2.30.3",
  "libsdl2 2.30.4",
  "libsdl2 2.30.5",
  "libsdl2 2.30.6",
  "libsdl2 2.30.7",
  "libsdl2 2.30.8",
  "libsdl2 2.30.9",
  "libsdl2 2.30.10",
  "libsdl2_gfx",
  "libsdl2_gfx 1.0.4",
  "libsdl2_image",
  "libsdl2_image 2.8.4",
  "libsdl2_image 2.8.3",
  "libsdl2_image 2.6.0",
  "libsdl2_image 2.6.1",
  "libsdl2_image 2.6.2",
  "libsdl2_image 2.6.3",
  "libsdl2_image 2.8.0",
  "libsdl2_image 2.8.1",
  "libsdl2_image 2.8.2",
  "libsdl2_mixer",
  "libsdl2_mixer 2.0.4",
  "libsdl2_mixer 2.6.0",
  "libsdl2_mixer 2.6.1",
  "libsdl2_mixer 2.6.2",
  "libsdl2_mixer 2.8.0",
  "libsdl2_net",
  "libsdl2_net 2.2.0",
  "libsdl2_ttf",
  "libsdl2_ttf 2.20.0",
  "libsdl2_ttf 2.20.1",
  "libsdl2_ttf 2.20.2",
  "libsdl2_ttf 2.22.0",
  "libsdl_gfx",
  "libsdl_image",
  "libsdl_mixer",
  "libsdl_net",
  "libsdl_ttf",
  "magnum",
  "openscenegraph",
  "pdcurses",
  "pdcursesmod",
}
```
there's a high chance at least one of the CI job will fail but this can be safely merged as it's just a package naming.

waruqi commented:
> there's a high chance at least one of the CI job will fail but this can be safely merged as it's just a package naming.

All ci failed. It was very risky to merge so many changes at once.

I think it should at least be split into several patches, tested and merged separately.

patch 1: rename libsdl -> libsdl2, use set_base in libsdl
patch 2: rename libsdl_image -> libsdl2_image, , use set_base in libsdl_image
patch 3: rename libsdl2_mixer, ..
patch 4: rename libsdl2_ttf, ..
...
last patch: rename add_deps() in other packages

SirLynix commented:
I split it into three commits:
- [one renaming the packages](https://github.com/xmake-io/xmake-repo/pull/6292/commits/7912b7dead8879be5284e6bcecc8d0867c6c6346) (so you can see what I changed in the package)
- [one re-adding libsdl stubs to avoid breaking projects](https://github.com/xmake-io/xmake-repo/pull/6292/commits/62d9241cc6deb3579cbdf537b232bbc2109ff514)
- [one updating existing deps in xmake-repo from libsdl to libsdl2](https://github.com/xmake-io/xmake-repo/pull/6292/commits/bf8b77ad85714d0f234505a64727abcb2c6f7819)

CI will fail because it tries to compile every known version of libsdl2, some broke due to package updates, other were never tested on platforms that were enabled later, etc.

So yeah I think this PR has to be manually reviewed, hopefully the commits diff are quite easy to read

waruqi commented:
> I split it into three commits:
> 
> * [one renaming the packages](https://github.com/xmake-io/xmake-repo/pull/6292/commits/7912b7dead8879be5284e6bcecc8d0867c6c6346) (so you can see what I changed in the packages)
> * [one re-adding libsdl stubs to avoid breaking projects](https://github.com/xmake-io/xmake-repo/pull/6292/commits/62d9241cc6deb3579cbdf537b232bbc2109ff514)
> * [one updating existing deps in xmake-repo from libsdl to libsdl2](https://github.com/xmake-io/xmake-repo/pull/6292/commits/bf8b77ad85714d0f234505a64727abcb2c6f7819)
> 
> CI will fail because it tries to compile every known version of libsdl2, some broke due to package updates, other were never tested on platforms that were enabled later, etc.
> 
> So yeah I think this PR has to be manually reviewed, hopefully the commits diff are quite easy to read

But all in one PR, it is too big, hard to review, and always causes CI to test all versions and packages. I want to split them into three PRs

SirLynix commented:
I don't see the point, it's easy to review each commit separately.
CI will fail even if I split it in three PR because it will try to test all versions of libsdl on all platforms.

Also if I split it, there will be a possibility for people to get xmake-repo with an incomplete migration (for example if we merge libsdl2 rename but not imgui, people using imgui will get an error because it will use libsdl)

waruqi commented:
> CI will fail even if I split it in three PR because it will try to test all versions of libsdl on all platforms, or all projects using it (johnnyengine doesn't pass the CI for example)

you can edit test.lua to only test latest version. if ci tests are passed, we can revert this patch.

> https://github.com/xmake-io/xmake-repo/blob/d7a0779921ae2ce42700398f1ecaa0995fca491b/scripts/test.lua#L275

> Also if I split it, there will be a possibility for people to get xmake-repo with an incomplete migration (for example if we merge libsdl2 rename but not imgui dep update, people using imgui will get a warning because it will use libsdl)

pr 1: libsdl -> libsdl2, libsdl set_base + warnings.

It will not break anything, only one warnings. user still can use `add_requires("libsdl")`.

In addition, even if pr 1 is merged, it will not be synchronized to master immediately, so we still have time to submit pr 2

SirLynix commented:
That's still quite a lot of work for something as simple as renaming a package, I don't have time right now to do it.

waruqi commented:
pr1: https://github.com/xmake-io/xmake-repo/pull/6307

waruqi commented:
pr2: https://github.com/xmake-io/xmake-repo/pull/6316

waruqi commented:
pr3: https://github.com/xmake-io/xmake-repo/pull/6317

waruqi commented:
this is the last pr.

SirLynix commented:
merging https://github.com/xmake-io/xmake/pull/6145 should be enough to fix compilation on arm64
